modules/file-system/filestore/outputs.tf

```diff
@@ -50,3 +50,13 @@ output "mount_runner" {
   EOT
   value       = local.mount_runner
 }
+
+output "filestore_id" {
+  description = "An identifier for the resource with format `projects/{{project}}/locations/{{location}}/instances/{{name}}`"
+  value       = google_filestore_instance.filestore_instance.id
+}
+
+output "capacity_gb" {
```

Comment:
The way I read the the [filestore] resource docs, the `file_shares` arguments are part of creation. I am not 100% certain, but this may cause Terraform to order resources incorrectly unless they explicitly depend upon the filestore resource itself. This might need a `depends_on` block.

[filestore]: https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/filestore_instance#file_shares

Reply:
I tested this and the `capacity_gb` output does not release until the filestore is created. 

I used the following blueprint to test:

```
- group: primary
  modules:
  - id: network1
    source: modules/network/pre-existing-vpc

  - id: homefs
    source: ./modules/file-system/filestore
    use: [network1]
    settings:
      local_mount: /home

  - id: wait1
    source: community/modules/scripts/wait-for-startup
    settings:
      instance_name: ((tostring(module.homefs.capacity_gb)))

  - id: wait2
    source: community/modules/scripts/wait-for-startup
    settings:
      instance_name: ((tostring(module.homefs.test)))
```

Where `test` is an output I added to filestore:

```
output "test" {
  value       = "a-string"
}
```

Terraform output:

```
module.homefs.random_id.resource_name_suffix: Creating...
module.homefs.random_id.resource_name_suffix: Creation complete after 0s [id=bredWg]
module.wait2.null_resource.wait_for_startup: Creating...
module.wait2.null_resource.wait_for_startup: Provisioning with 'local-exec'...
module.wait2.null_resource.wait_for_startup (local-exec): Executing: ["/bin/sh" "-c" "/bin/bash modules/embedded/community/modules/scripts/wait-for-startup/scripts/wait-for-startup-status.sh"]
module.homefs.google_filestore_instance.filestore_instance: Creating...
module.wait2.null_resource.wait_for_startup (local-exec): ERROR: (gcloud.compute.instances.get-serial-port-output) Could not fetch serial port output: The resource ... was not found
module.homefs.google_filestore_instance.filestore_instance: Still creating... [10s elapsed]
module.homefs.google_filestore_instance.filestore_instance: Still creating... [20s elapsed]
module.homefs.google_filestore_instance.filestore_instance: Still creating... [30s elapsed]
module.homefs.google_filestore_instance.filestore_instance: Still creating... [40s elapsed]
module.homefs.google_filestore_instance.filestore_instance: Still creating... [50s elapsed]
module.homefs.google_filestore_instance.filestore_instance: Still creating... [1m0s elapsed]
module.homefs.google_filestore_instance.filestore_instance: Still creating... [1m10s elapsed]
module.homefs.google_filestore_instance.filestore_instance: Still creating... [1m20s elapsed]
module.homefs.google_filestore_instance.filestore_instance: Still creating... [1m30s elapsed]
module.homefs.google_filestore_instance.filestore_instance: Still creating... [1m40s elapsed]
module.homefs.google_filestore_instance.filestore_instance: Still creating... [1m50s elapsed]
module.homefs.google_filestore_instance.filestore_instance: Still creating... [2m0s elapsed]
module.homefs.google_filestore_instance.filestore_instance: Still creating... [2m10s elapsed]
module.homefs.google_filestore_instance.filestore_instance: Creation complete after 2m14s [id=...]
module.wait1.null_resource.wait_for_startup: Creating...
module.wait1.null_resource.wait_for_startup: Provisioning with 'local-exec'...
module.wait1.null_resource.wait_for_startup (local-exec): Executing: ["/bin/sh" "-c" "/bin/bash modules/embedded/community/modules/scripts/wait-for-startup/scripts/wait-for-startup-status.sh"]
module.wait1.null_resource.wait_for_startup (local-exec): ERROR: (gcloud.compute.instances.get-serial-port-output) Could not fetch serial port output: The resource ...' was not found
```

You can see that the string is released immediately but capacity does not release until filestore creation is complete.